src/reducers/entities/posts.js

```diff
@@ -90,6 +90,35 @@ function handleReceivedPosts(posts = {}, postsInChannel = {}, action) {
     return {posts: nextPosts, postsInChannel: nextPostsForChannel};
 }
 
+function handlePostsFromSearch(posts = {}, postsInChannel = {}, action) {
+    const newPosts = action.data.posts;
+    let info = {posts, postsInChannel};
+    const postsForChannel = new Map();
+
+    for (const id in newPosts) {
+        if (newPosts.hasOwnProperty(id)) {
+            const nextPost = newPosts[id];
+            if (nextPost) {
+                const channelId = nextPost.channel_id;
+                if (postsForChannel.has(channelId)) {
+                    postsForChannel.set(channelId,
+                        {...postsForChannel.get(channelId), [id]: nextPost}
+                    );
+                } else {
+                    postsForChannel.set(channelId, {[id]: nextPost});
+                }
+            }
+        }
+    }
+
+    postsForChannel.forEach((postList, channelId) => {
+        const result = handleReceivedPosts(info.posts, postsInChannel, {channelId, data: {posts: postList}});
```

Comment:
not really cause this is for every channel meaning that we need to concat each result into info, does it make sense?